python/mxnet/optimizer.py

```diff
@@ -356,6 +358,52 @@ def update(self, index, weight, grad, state):
                        lr=lr, wd=wd, **kwargs)
 
 @register
+class MultiPrecisionSGD(Optimizer):
```

Comment:
Would you like it to be just an additional option to sgd then, like that:
precision: "same" (keep the precision of input, default), "float32"?